ShareX.HelpersLib/Input/MouseClickEffectForm.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Drawing;
+using System.Windows.Forms;
+
+namespace ShareX.HelpersLib
+{
+    // transparent form where mouse effect will be drawn
+    public class MouseClickEffectForm : Form
+    {
+        private bool _drawEffect = false;
+        public MouseClickEffectForm()
+        {
+            Size = new Size(40, 40);
+            FormBorderStyle = FormBorderStyle.None;
+            ShowIcon = false;
+            ShowInTaskbar = false;
+            TopMost = true;
+            AllowTransparency = true;
+            BackColor = Color.White; // Set a key color for transparency
+            TransparencyKey = Color.White; // Make that color transparent
+        }
+
+        /// <summary>
+        /// Draw mouse effect on given mouse position
+        /// </summary>
+        public void DrawMouseEffect(Point cursorPosition)
+        {
+            _drawEffect = true;
+            CenterFormToCursorPosition(cursorPosition);
+            Invalidate(); // redraw form
+        }
+
+        /// <summary>
+        /// Clear mouse effect
+        /// </summary>
+        public void ClearMouseEffect()
+        {
+            _drawEffect = false;
+            Invalidate(); // redraw form
+        }
+
+        /// <summary>
+        /// Draw a hollow circle as an mouse effect
+        /// </summary>
+        protected override void OnPaint(PaintEventArgs e)
+        {
+            if (_drawEffect)
+            {
+                Graphics g = e.Graphics;
+                g.SmoothingMode = System.Drawing.Drawing2D.SmoothingMode.AntiAlias;
+                // Define a pen to draw hollow circle
+                Pen pen = new Pen(Color.Red, 3);
+                int diameter = 10;
+
+                // Calculate the top-left corner to center the circle
+                int x = (ClientSize.Width - diameter) / 2;
+                int y = (ClientSize.Height - diameter) / 2;
+
+                // .NET GDI+ is not precise when drawign circles this would be better off with WPF/vector-based drawing
+                g.DrawEllipse(pen, x, y, diameter, diameter);
```

Comment:
This might not be best solution due to GDI+ poor precision.
![ShareX_CwfyI1M3JM](https://github.com/user-attachments/assets/9a01b826-5034-4841-8a21-5065a25dc04c)

Reply:
I can't add that much to this, I just wanted to say that rn I'm using this tool https://github.com/Phaiax/Key-n-Stroke to achieve the same and as it is open-source already you guys can probably get inspiration from there and/or took over some of the code to not reinvent the wheel here :) (altho I dunno if the poor precision thing is really better with keynstroke --> at least from a enduser-experience I've never had simliar problems)

Reply:
Good point @saintcore. I do get inspired by projects like this. I like keyboard capturing capability it has and I might take a stab at it after I'm done with mouse effect.

I don't know how things work regarding "taking over" some code in the open-source world. I guess some collaboration with the owner (@Phaiax) is needed.

Regarding the small issue I mentioned here, I think that might be solved with help from @Jaex but keynstroke uses WPF so it is slightly different than ShareX which uses Windows Forms.

Reply:
Well according to the provided license, it shouldn't be a problem (see https://github.com/Phaiax/Key-n-Stroke?tab=Apache-2.0-1-ov-file#readme ). A good summary of what apache 2.0 license allows us to do with and not is available over at snyk: https://snyk.io/learn/apache-license/

> Under the Apache license, users are permitted to modify, distribute, and sublicense the original open source code. Commercial use, warranties, and patent claims are also allowed with the Apache license. 
....
When using the Apache license, developers must include the original copyright notice, a copy of the license text itself, and in some cases, a copy of the notice file with attribution notes and a disclosure of any significant changes made to the original code.
....

With that said, ofc it would be nice to hear from the original author of keynstroke and his desires beside license thoughts :)